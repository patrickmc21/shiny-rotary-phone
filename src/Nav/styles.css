.App-Nav {
  height: 80px;
}