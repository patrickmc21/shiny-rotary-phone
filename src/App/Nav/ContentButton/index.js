import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentButton = () => {
  return(
    <div>I'm a content button</div>
  )
}

export default ContentButton;