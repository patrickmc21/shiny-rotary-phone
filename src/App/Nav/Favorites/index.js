import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Favorites = () => {
  return(
    <div>I'm a Favorites button</div>
  )
}

export default Favorites;