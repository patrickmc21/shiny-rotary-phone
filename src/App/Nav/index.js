import React from 'react';
import ContentButton from './ContentButton/index.js';
import Favorites from './Favorites/index.js';
import PropTypes from 'prop-types';
import './styles.css';

const Nav = () => {
  return (
    <nav className='App-Nav'>
      <ContentButton buttonType='people'/>
      <ContentButton buttonType='planets'/>
      <ContentButton buttonType='vehicles'/>
      <Favorites />
    </nav>
  )
}

export default Nav;