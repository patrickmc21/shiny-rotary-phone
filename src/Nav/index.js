import React from 'react';
import { Link } from 'react-router-dom';
import ContentButton from '../ContentButton/index.js';
import Favorites from '../Favorites/index.js';
import PropTypes from 'prop-types';
import './styles.css';

const Nav = ({activateCategory, buttonType, numberOfFavorites, activeCategoryName}) => {
  const buttonsWithoutFavs = buttonType.filter(button => button !== 'favorites');
  const buttons = buttonsWithoutFavs.map(button => {
    return (
      <Link to={`/${button}`} key={button}>
          <ContentButton 
            buttonType={button}
            activateCategory={activateCategory}
            activeCategoryName={activeCategoryName} />
      </Link>
    )
  })
  return (
    <nav className='App-Nav'>
      {buttons}
      <Link to='/favorites'>
        <Favorites 
          numberOfFavorites={numberOfFavorites}
          activateCategory={activateCategory}
          activeCategoryName={activeCategoryName}/>
      </Link>
    </nav>
  )
}

Nav.propTypes = {
  activateCategory: PropTypes.func,
  buttonType: PropTypes.array,
  numberOfFavorites: PropTypes.number
}

export default Nav;