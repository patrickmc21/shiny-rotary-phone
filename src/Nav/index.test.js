import React from 'react';
import Nav from './index.js';
import { shallow } from 'enzyme';

describe('Nav', () => {

  let wrapper;
  let mockActivateCategory;
  let mockButtonType;
  let mockNumberOfFavorites;

  beforeEach(() => {
    mockNumberOfFavorites = 3;
    mockActivateCategory = jest.fn();
    mockButtonType = ['people', 'vehicles', 'planets'];
    wrapper = shallow(
      <Nav 
        activateCategory={mockActivateCategory}
        buttonType={mockButtonType}
        numberOfFavorites={mockNumberOfFavorites}/>);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});