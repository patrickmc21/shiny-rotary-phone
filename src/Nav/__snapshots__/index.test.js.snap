// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    activateCategory={[Function]}
    buttonType={
        Array [
            "people",
            "vehicles",
            "planets",
          ]
    }
    numberOfFavorites={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Link
            replace={false}
            to="/people"
>
            <ContentButton
                        activateCategory={[Function]}
                        activeCategoryName={undefined}
                        buttonType="people"
            />
</Link>,
          <Link
            replace={false}
            to="/vehicles"
>
            <ContentButton
                        activateCategory={[Function]}
                        activeCategoryName={undefined}
                        buttonType="vehicles"
            />
</Link>,
          <Link
            replace={false}
            to="/planets"
>
            <ContentButton
                        activateCategory={[Function]}
                        activeCategoryName={undefined}
                        buttonType="planets"
            />
</Link>,
        ],
        <Link
          replace={false}
          to="/favorites"
>
          <Favorites
                    activateCategory={[Function]}
                    activeCategoryName={undefined}
                    numberOfFavorites={3}
          />
</Link>,
      ],
      "className": "App-Nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "people",
        "nodeType": "class",
        "props": Object {
          "children": <ContentButton
            activateCategory={[Function]}
            activeCategoryName={undefined}
            buttonType="people"
/>,
          "replace": false,
          "to": "/people",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activateCategory": [Function],
            "activeCategoryName": undefined,
            "buttonType": "people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "vehicles",
        "nodeType": "class",
        "props": Object {
          "children": <ContentButton
            activateCategory={[Function]}
            activeCategoryName={undefined}
            buttonType="vehicles"
/>,
          "replace": false,
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activateCategory": [Function],
            "activeCategoryName": undefined,
            "buttonType": "vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "planets",
        "nodeType": "class",
        "props": Object {
          "children": <ContentButton
            activateCategory={[Function]}
            activeCategoryName={undefined}
            buttonType="planets"
/>,
          "replace": false,
          "to": "/planets",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activateCategory": [Function],
            "activeCategoryName": undefined,
            "buttonType": "planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Favorites
            activateCategory={[Function]}
            activeCategoryName={undefined}
            numberOfFavorites={3}
/>,
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activateCategory": [Function],
            "activeCategoryName": undefined,
            "numberOfFavorites": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Link
              replace={false}
              to="/people"
>
              <ContentButton
                            activateCategory={[Function]}
                            activeCategoryName={undefined}
                            buttonType="people"
              />
</Link>,
            <Link
              replace={false}
              to="/vehicles"
>
              <ContentButton
                            activateCategory={[Function]}
                            activeCategoryName={undefined}
                            buttonType="vehicles"
              />
</Link>,
            <Link
              replace={false}
              to="/planets"
>
              <ContentButton
                            activateCategory={[Function]}
                            activeCategoryName={undefined}
                            buttonType="planets"
              />
</Link>,
          ],
          <Link
            replace={false}
            to="/favorites"
>
            <Favorites
                        activateCategory={[Function]}
                        activeCategoryName={undefined}
                        numberOfFavorites={3}
            />
</Link>,
        ],
        "className": "App-Nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "people",
          "nodeType": "class",
          "props": Object {
            "children": <ContentButton
              activateCategory={[Function]}
              activeCategoryName={undefined}
              buttonType="people"
/>,
            "replace": false,
            "to": "/people",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activateCategory": [Function],
              "activeCategoryName": undefined,
              "buttonType": "people",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "vehicles",
          "nodeType": "class",
          "props": Object {
            "children": <ContentButton
              activateCategory={[Function]}
              activeCategoryName={undefined}
              buttonType="vehicles"
/>,
            "replace": false,
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activateCategory": [Function],
              "activeCategoryName": undefined,
              "buttonType": "vehicles",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "planets",
          "nodeType": "class",
          "props": Object {
            "children": <ContentButton
              activateCategory={[Function]}
              activeCategoryName={undefined}
              buttonType="planets"
/>,
            "replace": false,
            "to": "/planets",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activateCategory": [Function],
              "activeCategoryName": undefined,
              "buttonType": "planets",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Favorites
              activateCategory={[Function]}
              activeCategoryName={undefined}
              numberOfFavorites={3}
/>,
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activateCategory": [Function],
              "activeCategoryName": undefined,
              "numberOfFavorites": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
